infrastructure/src/main/java/org/corfudb/infrastructure/ResourceQuota.java

```diff
@@ -0,0 +1,43 @@
+package org.corfudb.infrastructure;
+
+import lombok.AllArgsConstructor;
+import org.corfudb.runtime.exceptions.QuotaExceededException;
+
+
+/**
+ * A quota for a resource that can be modeled with a long.
+ *
+ * Created by Maithem on 6/20/19.
+ */
+@AllArgsConstructor
+public class ResourceQuota {
+
+    private final String name;
+    private final long used;
+    private final long limit;
+
+    /**
+     * Consumes an amount of the resource quota.
+     */
+    public ResourceQuota acquire(long amount) {
+        return new ResourceQuota(name, used + amount, limit);
+    }
+
+    /**
+     * Checks if the resource quota has been exceeded.
+     */
+    public void check() {
+        if (used > limit) {
+            String msg = String.format("Resource Exhausted[%s]: limit=%d, exceededSize=%d",
```

Comment:
exceeded size suggests it's how much size was exceeded, used - limit? maybe renaming to be more clear.